src/openms/include/OpenMS/COMPARISON/CLUSTERING/HashGrid.h

```diff
@@ -405,16 +401,28 @@ namespace OpenMS
 
     /**
      * @warning Currently needed non-const by HierarchicalClustering.
-     */
-    grid_iterator grid_begin() { return cells_.begin(); }
+    */
+    typename Grid::mapped_type & grid_at(const CellIndex & x) { return cells_.at(x); }
```

Comment:
can this be const (as well as the other grid_finds)

Reply:
this one yes.  other grid_find, no, otherwise you cannot overload since only the return type is different.

Reply:
ah no, actually none of them.

Reply:
both have colliding overloads